other/coordinate_compression.py

```diff
@@ -0,0 +1,141 @@
+"""
+Assumption:
+    - The values to compress are assumed to be comparable,
+    - values can be sorted and compared with '<' and '>' operators.
+"""
+
+
+class CoordinateCompressor:
+    """
+    A class for coordinate compression.
+
+    This class allows you to compress and decompress a list of values.
+
+    Mapping:
+    In addition to compression and decompression, this class maintains a mapping
+    between original values and their compressed counterparts using two data
+    structures: a dictionary `coordinate_map` and a list `reverse_map`.
+
+    - `coordinate_map`: A dictionary that maps original values to their compressed
+      coordinates. Keys are original values, and values are compressed coordinates.
+
+    - `reverse_map`: A list used for reverse mapping, where each index corresponds
+      to a compressed coordinate, and the value at that index is the original value.
+
+    Example mapping:
+
+    Original: 10, Compressed: 0
+    Original: 52, Compressed: 1
+    Original: 83, Compressed: 2
+    Original: 100, Compressed: 3
+
+    This mapping allows for efficient compression and decompression of values within
+    the list.
+    """
+
+    def __init__(self, arr: list[int | float | str]) -> None:
+        """
+        Initialize the CoordinateCompressor with a list.
+
+        Args:
+        arr (list): The list of values to be compressed.
```

Comment:
Mypy can check the data types in the function signature but not in the comments.

Please do not repeat the data types in both places because readers will be confused if one changes but the other does not.
```suggestion
        arr: The list of values to be compressed.
```